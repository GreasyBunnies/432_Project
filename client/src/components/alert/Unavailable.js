import React from 'react';
import {
    Alert
} from 'react-bootstrap';

class Unavailable extends React.Component {
    render() {
        return (
            <Alert variant="danger">
                <Alert.Heading>Service Unavailable</Alert.Heading>
                <p>{this.props.message}</p>
            </Alert>
        )
    }
}

export default Unavailable;
