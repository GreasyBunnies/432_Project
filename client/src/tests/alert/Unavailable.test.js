import React from 'react';
import ReactDOM from 'react-dom';

import Unavailable from '../../components/alert/Unavailable.js';

import 'jest-localstorage-mock';
import renderer from 'react-test-renderer';

describe("Rendering Unavailable alert", () => {
	it('Renders Unavailable without crashing', () => {
		const div = document.createElement('div');

		ReactDOM.render(<Unavailable />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
});

describe("Rendering Unavailable alert with message", () => {
	it('Displays the correct message', () => {
		const component = renderer.create(
			<Unavailable message="This is a test"></Unavailable>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
