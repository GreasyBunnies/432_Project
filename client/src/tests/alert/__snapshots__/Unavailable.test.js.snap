// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Unavailable alert with message Displays the correct message 1`] = `
<div
  className="fade alert alert-danger show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    Service Unavailable
  </div>
  <p>
    This is a test
  </p>
</div>
`;
